### PR TITLE
Small changes

### PR DESCRIPTION
**Pull request by delba**
Hi Luke,

I never would have dreamt that a lib would be created on top on `Permission` 😍😍
Thank you so much!!!

I tried to give back a bit with this PR ;)

### CODE REVIEW

File: Source/RxPermission.swift [hunk 1]
```diff
@@ -1,52 +1,85 @@
...
+    public var rx_permission: Observable<PermissionStatus> {
         return rx_permissionInstance.asObservable()
     }
-    
-    private var rx_permissionInstance: PublishSubject<PermissionStatus> {
+}
+
+// MARK: - Permission
+
+private extension Permission: AssociatedObject {
```
Comment (sunshinejr) [inline Source/RxPermission.swift]: We can't use private with protocol conformances 😢 I'd say we just move private to functions below?

File: Source/RxPermission.swift [hunk 2]
```diff
@@ -1,52 +1,85 @@
...
+    var rx_permissionInstance: PublishSubject<PermissionStatus> {
         get {
-            var permission: PublishSubject<PermissionStatus>!
+            let permission: PublishSubject<PermissionStatus>
+            
             doLocked {
-                let lookup = objc_getAssociatedObject(self, &AssociatedKeys.Permission) as? PublishSubject<PermissionStatus>
-                if let lookup = lookup {
+                if let lookup = associatedObject(&.permission) as? PublishSubject<PermissionStatus> {
```
Comment (sunshinejr) [inline Source/RxPermission.swift]: We have to use `self.associatedObject()` because it is a closure 😢

File: Source/RxPermission.swift [hunk 3]
```diff
@@ -1,52 +1,85 @@
...
                 }
             }
+            
             return permission
         }
         
         set {
             doLocked {
-                objc_setAssociatedObject(self, &AssociatedKeys.Permission, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+                associatedObject(&.permission, newValue)
```
Comment (sunshinejr) [inline Source/RxPermission.swift]: Same here, `self` here and you forgot about `object:` label in second parameter

**Comment by sunshinejr** [comment]
Hey Damien! 🎉

First of all thanks for your kind words! I really love `Permission`, you did great job on that! 💯 As for pull request, it seems also really nice, thank you! I've made some comments but that shouldn't be a problem 🐼

**Comment by delba** [comment]
😱 Sorry for all the errors, I coded without the compiler...
I rebased the PR, it should be fine now ;)

BTW do you plan to support Carthage ?

**Comment by sunshinejr** [comment]
No worries 🐼 It looks perfect now, thanks again! About Carthage, of course, just I'm not _that_ experienced with Cathage so wanted to do it later, but if you have any experience, I would be more than happy to see a PR from you :)

**Comment by sunshinejr** [comment]
I've also added you as a collaborator to this project, if you don't mind 🎉

**Comment by delba** [comment]
Ahah great thanks!

**Pull request merged**